### DISCUSSION TITLE
Memory leak in HttpHeader.h

### DISCUSSION THREAD

**Discussion by pete8804** [pschatzmann/arduino-audio-tools#1088] [Q&A] [answered]
After updating my web radio project to the most recent version of arduino_audio_tools, 
I noticed that my project was consuming ca. 4k of memory on each restart of an URL stream. 
After some investigation I was able to identify the cause:
On 2023-09-22, method **clear() in HttpRequest.h** was changed as follows:
 ```
 /// clears the data - usually we do not delete but we just set the active flag
        HttpHeader& clear(bool activeFlag=true) {
            is_written = false;
            is_chunked = false;
            url_path = "/";
            for (auto it = lines.begin() ; it != lines.end(); ++it){
                if (activeFlag){
                    (*it)->active = false;
                } else {
                    (*it)->active = false;  <- added 2023-09-22
                    //delete *it;  // this leads to exceptions <- commented out 2023-09-22
                }
            }
            if (!activeFlag){
                lines.clear(); // BUG: memory leak if delete *it is commented out
            }
            return *this;
```            
When _lines.clear()_ is executed without preceding _delete *it,_ the existing header lines become inaccessible, but memory is not released. Then, on the next start of an URL stream, new header lines are created instead of re-activating the existing ones.

I fixed the problem locally by commenting out the _lines.clear()_ statement, and successfully retested it.
I hope you can confirm my observation and implement the fix.

By the way: arduino_audio_tools is a great piece of work. I really appreciate it!

**Comment by pschatzmann** [comment, marked as answer]
Converted to an [issue](https://github.com/pschatzmann/arduino-audio-tools/issues/1097)
